android/app/src/main/kotlin/coop/polypoly/polypod/MainActivity.kt

```diff
@@ -61,101 +51,15 @@ class MainActivity : AppCompatActivity() {
                 }
                 .show()
         }
-    }
-
-    fun authorize(successfulAuth: (() -> Unit)) {
-        ensureLockScreen(successfulAuth) {
-            val promptInfo = BiometricPrompt.PromptInfo.Builder()
-                .setTitle(this.getString(R.string.auth_title))
-                .setSubtitle(this.getString(R.string.auth_subtitle))
-                .setAllowedAuthenticators(desiredLockScreenType)
-                .build()
-
-            val executor = ContextCompat.getMainExecutor(this)
-            val callback = PolyAuthCallback(this, successfulAuth)
-
-            BiometricPrompt(this, executor, callback).authenticate(promptInfo)
-        }
-    }
-
-    fun ensureLockScreen(
-        noScreenLock: (() -> Unit),
-        biometricAuth: (() -> Unit)
-    ) {
-        if (!Preferences.getBiometricCheck(this)) {
-            return noScreenLock()
-        }
-        val biometricManager = BiometricManager.from(this)
-        if (biometricManager.canAuthenticate(
-                desiredLockScreenType
-            ) != BiometricManager.BIOMETRIC_ERROR_NONE_ENROLLED
+        if (Preferences.isFirstRun(this)
```

Comment:
Why check isFirstRun twice? I found this clearer in the same condition that marks the notification as read, doesn't seem to actually affect semantics.